js/main.js

```diff
@@ -1,17 +1,60 @@
-// Функция возвращающая случайное целое число из переданного диапазона включительно.
+const features = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+const photos = [
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg',
+  'https://assetshtmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg',
+];
 
-const randomInteger = function (min, max) {
-  // случайное число от min до (max+1)
-  const rand = min + Math.random() * (max + 1 - min);
-  return Math.floor(rand);
+const getRandomPositiveInteger = (min, max) => {
+  const lower = Math.ceil(Math.min(Math.abs(min), Math.abs(max)));
+  const upper = Math.floor(Math.max(Math.abs(min), Math.abs(max)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
 };
-randomInteger(0, 100);
 
-// Функция, возвращающая случайное число с плавающей точкой из переданного диапазона включительно.
+const getRandomPositiveFloat = (min, max, digits = 1) => {
+  const lower = Math.ceil(Math.min(Math.abs(min), Math.abs(max)));
+  const upper = Math.max(Math.abs(min), Math.abs(max));
+  const result = Math.random() * (upper - lower) + lower;
+  return result.toFixed(digits);
+};
+
+// Функция копирует, затем перемешивает массив в случайном порядке
+const getShuffledItems = (items) => items.slice().sort(() => Math.random() - 0.5);
+
+// Функция возвращает перемешенный массив до заданной длины массива
+const getRandomItems = (items, count) => {
+  const shuffledItems = getShuffledItems(items);
+  return shuffledItems.slice(0, count);
+};
+
+const getRandomLat = getRandomPositiveFloat(35.65, 35.7, 5);
+const getRandomLng = getRandomPositiveFloat(35.65, 35.7, 5);
 
-const getFractionalNumber = function (min, max, fixed) {
-  const randomNumber = Math.random() * (max + 1 - min) + min;
-  return Number(randomNumber.toFixed(fixed));
+// Создает новый объект с рандомными свойствами
+const createAuthor = function () {
+  return {
+    avatar: `img/avatars/user0${getRandomPositiveInteger(1, 8)}.png`,
+    offer: {
+      title: 'Добро пожаловать',
+      address: `${getRandomLat}, ${getRandomLng}`,
+      price: getRandomPositiveInteger(1000, 10000),
```

Comment:
Сейчас по всему коду разбросаны значения для генерации моков. Давай их все соберем в одном месте
Что-то вроде:
```js
const MocksConfig = {
  OFFERS_COUNT: 10,
  AVATAR: {
    MIN: 1,
    MAX: 8,
 },
  PRICE: {
    MIN: 1000,
    MAX: 10000, 
 },
};
```
И потом использовать его в коде
```suggestion
    avatar: `img/avatars/user0${getRandomPositiveInteger(MocksConfig.AVATAR.MIN, MocksConfig.AVATAR.MAX)}.png`,
    offer: {
      title: 'Добро пожаловать',
      address: `${getRandomLat}, ${getRandomLng}`,
      price: getRandomPositiveInteger(MocksConfig.PRICE.MIN, MocksConfig.AVATAR.MAX),
```
И так по со всеми данными для генерации

У нас все будет в одном месте, что очень упрощает жизнь